ext/groonga/rb-grn-database.c

```diff
@@ -623,6 +624,60 @@ rb_grn_database_unmap (VALUE self)
     return Qnil;
 }
 
+/*
+ * Recreates all index columns in the database.
+ *
+ * This method is useful when you have any broken index columns in
+ * the database. You don't need to specify each index column. But
+ * this method spends more time rather than you specify only reindex
+ * needed index columns.
+ *
+ * @example How to recreate all index columns in the database
+ *   database = Groonga::Database.create(:path => "/tmp/db")
+ *
+ *   Groonga::Schema.define do |schema|
+ *     schema.create_table("Memos") do |table|
+ *       table.short_text("title")
+ *       table.text("content")
+ *     end
+ *
+ *     schema.create_table("Terms",
+ *                         :type => :patricia_trie,
+ *                         :key_type => :short_text,
+ *                         :normalizer => "NormalizerAuto",
+ *                         :default_tokenizer => "TokenBigram") do |table|
+ *       table.index("Memos.title")
+ *       table.index("Memos.content")
+ *     end
+ *   end
+ *
+ *   database.reindex
+ *   # =>
+ *   #   Groonga["Terms.Memos_title"].reindex
+ *   #   Groonga["Terms.Memos_content"].reindex
+ *
```

Comment:
Sorry. The example I suggested isn't best. Sample schema should have two or more lexicons to show difference with `table#reindex`:

```ruby
database = Groonga::Database.create(:path => "/tmp/db")

Groonga::Schema.define do |schema|
  schema.create_table("Memos") do |table|
    table.short_text("title")
    table.text("content")
  end

  schema.create_table("BigramTerms",
                      :type => :patricia_trie,
                      :key_type => :short_text,
                      :normalizer => "NormalizerAuto",
                      :default_tokenizer => "TokenBigram") do |table|
    table.index("Memos.title")
    table.index("Memos.content")
  end

  schema.create_table("MeCabTerms",
                      :type => :patricia_trie,
                      :key_type => :short_text,
                      :normalizer => "NormalizerAuto",
                      :default_tokenizer => "TokenMecab") do |table|
    table.index("Memos.title")
    table.index("Memos.content")
  end
end

database.reindex
# =>
#   Groonga["BigramTerms.Memos_title"].reindex
#   Groonga["BigramTerms.Memos_content"].reindex
#   Groonga["MeCabTerms.Memos_title"].reindex
#   Groonga["MeCabTerms.Memos_content"].reindex
```